internal/indexmanagement/reconcile.go

```diff
@@ -94,18 +94,37 @@ func (imr *IndexManagementRequest) createOrUpdateIndexManagement() error {
 	spec := verifyAndNormalize(imr.cluster)
 	policies := spec.PolicyMap()
 
-	imr.cullIndexManagement(spec.Mappings, policies)
-	for _, mapping := range spec.Mappings {
-		ll := log.WithValues("mapping", mapping.Name)
-		// create or update template
-		if err := imr.createOrUpdateIndexTemplate(mapping); err != nil {
-			ll.Error(err, "failed to create index template")
-			return err
+	labels := map[string]string{
+		"cluster-name": imr.cluster.Name,
+		"component":    "elasticsearch",
+	}
+	esPods, err := pod.List(context.TODO(), imr.client, imr.cluster.Namespace, labels)
+	if err != nil {
+		return err
+	}
+
+	running := false
+	for _, pod := range esPods {
+		if pod.Status.Phase == corev1.PodRunning {
+			running = true
+			break
```

Comment:
why not use something like https://github.com/openshift/elasticsearch-operator/blob/master/internal/elasticsearch/elasticsearch.go#L31

Reply:
This is not possible because the two packages `indexmanagement` and `elasticsearch` would have a circular dependecy.